restapi/src/main/java/io/stargate/web/docsapi/exception/ErrorCodeRuntimeException.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.exception;
+
+import java.util.Objects;
+import javax.ws.rs.core.Response;
+
+/** {@link RuntimeException} that carries an {@link ErrorCode}. */
+public class ErrorCodeRuntimeException extends RuntimeException {
+
+  /** Dedicated {@link ErrorCode}. */
+  private final ErrorCode errorCode;
+
+  /** ErrorCode only constructor. */
+  public ErrorCodeRuntimeException(ErrorCode errorCode) {
+    this.errorCode = errorCode;
```

Comment:
Should we perhaps use `super(errorCode.defaultMessage())` in this case? Otherwise the exception will have a `null` message, AFAIK, which does not show up nicely in logs.

Reply:
Perfect, I can then further simplify in the `ErrorCode` I guess..

Reply:
Done.